File: Frontend/src/types/appointment.ts
export interface AvailableSlot {
  start: string;
  end: string;
  isAvailable: boolean;
  price?: number | null;
}

export interface BookAppointmentRequest {
  doctorId: number;
  scheduledAt: string;
  type: ConsultationType;
}

export interface AppointmentBooked {
  appointmentId: number;
  doctorId: number;
  scheduledAt: string;
  consultationType: ConsultationType;
  status: string;
  durationMins: number;
  price: number;
  paymentUrl?: string;
}

/** Matches backend ConsultationType enum: Chat=0, VideoCall=1, Clinic=2 */
export type ConsultationType = 0 | 1 | 2;

export const ConsultationType = {
  Chat: 0,
  VideoCall: 1,
  Clinic: 2,
} as const;

File: Backend/Tabibi.Core/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tabibi.Core.Models
{

    public class Appointment
    {
        [Key]
        public long AppointmentId { get; set; }

        public long PatientId { get; set; }
        public long DoctorId { get; set; }

        public DateTime ScheduledAt { get; set; }
        public int DurationMins { get; set; } = 30;

        [Required]
        public ConsultationType ConsultationType { get; set; } = ConsultationType.Chat;

        public AppointmentStatus Status { get; set; } = AppointmentStatus.Confirmed;

        [Column(TypeName = "decimal(10,2)")]
        public decimal Price { get; set; } = 0;

        public string? Notes { get; set; }

        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.Online;

        public long? SessionId { get; set; }

        // Navigation
        [ForeignKey(nameof(PatientId))]
        public PatientProfile Patient { get; set; } = null!;

        [ForeignKey(nameof(DoctorId))]
        public DoctorProfile Doctor { get; set; } = null!;

        [ForeignKey(nameof(SessionId))]
        public ChatSession? ChatSession { get; set; }

        public Payment? Payment { get; set; }
        public DoctorReview? Review { get; set; }
    }

    public enum ConsultationType { Chat = 0, VideoCall = 1, Clinic = 2 }
    public enum AppointmentStatus { Confirmed = 1, Completed = 2, Cancelled = 3, Pending = 4 }
    public enum PaymentMethod { Online = 1, OnSite = 2 }

}